quickref/qsharp-quick-reference.tex

```diff
@@ -139,14 +139,24 @@
   \end{keysref}
 
   \begin{keysref}{Derived Types}
-    Arrays                  & \texttt{\emph{elementType}[]} \\
+    Arrays                  & \texttt{\emph{elementType}[]} \newline
+    							\texttt{(elementType,elementType)[][] } \\
     Tuples                  & \texttt{(\emph{type0}, \emph{type1}, ...)} \newline
                               e.g.: \texttt{(Int, Qubit)} \\
     Functions               & \texttt{\emph{input} -> \emph{output}} \newline
                               e.g.: \texttt{ArcCos : (Double) -> Double} \\
     Operations              & \texttt{\emph{input} => \emph{output} : \emph{variants}} \newline
-                              e.g.: \texttt{H : (Qubit => Unit : \newline Adjoint, Controlled)} \\
+                              e.g.: \texttt{H : (Qubit => Unit : \newline Adj, Ctl)} \\
   \end{keysref}
+  
+  \begin{keysref}{User-Defined Types}
+  newtype  & \texttt{newtype Name = (Type, Type);} \newline
+            \texttt{e.g. newtype Complex = (Double, Double);} \\
+  nested newtype & \texttt{newtype Name1 = (type, (Name2 : type, type)); } \newline
+            \texttt{e.g. newtype Nested = (Double, (ItemName : Int, String));} \\
+  
+  \end{keysref}
+  
 
   \begin{keysref}{Functions, Operations and Types}
```

Comment:
I am not sure what you mean by remove types from this section. 

For the 2nd question, do you mean something like this where it uses the auto keyword?
`operation PrepareEntangledPair(here : Qubit, there : Qubit) : Unit {
    body (...) { // default body specialization
        H(here);
        CNOT(here, there);
    }

    adjoint auto; // auto-generate adjoint specialization
    controlled auto; // auto-generate controlled specialization
    controlled adjoint auto; // auto-generate controlled adjoint specialization
}`

Reply:
This section says "Functions, Operations and Types" and has an entry for defining a UDT. Now that we're creating a new section devoted solely to UDTs, we should call this section "Functions and Operations" and remove the entry for defining a UDT from it, since it would be duplicate

For the second comment, if you want to define adjoint and controlled variants for the operation manually, you have to spell out the whole `adjoint {...}` syntax; but if you want to define them automatically based on the body, you can say `adjoint auto` (as you correctly pointed out), or you can just say `operation PrepareEntangledPair(here : Qubit, there : Qubit) : Unit is Adj+Ctl`, and the compiler will figure out that it's a shorthand for `adjoint auto; controlled auto;`. This is a very convenient syntax, since it allows to skip `body(...)` part and save one level of nesting in the code.